tests/float/math_fun_special.py

```diff
@@ -50,3 +50,61 @@
             print("{:.4g}".format(function(value)))
         except ValueError as e:
             print(str(e))
+
+functions_var_args = [
+    (
+        "gcd",
+        gcd,
+        (
+            (1, 4),
+            (2, 8),
+            (0, 8),
+            (-1, 19),
+            (-5, 42),
+            (-7, -7),
+            (-9, -2),
+            (0, 0),
+            (6, 30, 40, -60, 20, 40),
+            (2147483643, 42),
+            (-2147483647, -214748364, 21474836),
+        ),
+    ),
+]
+
+for function_name, function, test_vals in functions_var_args:
+    print(function_name)
+    for values in test_vals:
+        try:
+            print("{:.4g}".format(function(*values)))
+        except ValueError as e:
+            print(str(e))
+
+print(gcd(12345678912345678912345678911561561658135153135135135, 123456))
+
+lcm_function_args = [
```

Comment:
it might bring more confidence to use a version of cpython's test cases for lcm see here: https://github.com/python/cpython/blob/main/Lib/test/test_math.py#L1062

I made similar changes to the gcd branch, https://github.com/ECE595/micropython/pull/11/files#diff-d7898171bbfbd9a3f017c3340ffe49519ffb629e9ac3c348577716b1ead032ccR55